content/kb/troubleshooting-some-old-instance-manager-e-are-still-running-after-upgrade.md

```diff
@@ -0,0 +1,31 @@
+---
+title: "Troubleshooting: Some old instance-manager in longhorn-system are still running after upgrade"
+author: Derek Su
+draft: false
+date: 2021-11-09
+categories:
+  - "Longhorn Upgrade"
+---
+
+## Applicable versions
+
+All Longhorn versions.
+
+## Symptoms
+
+Some old instance-manager in longhorn-system are still running after upgrade.
+
+## Details
+
+After upgrading the longhorn-manager, the user will see some old engine/replica instance manager pods (instance-manager-e/r) are still running in longhorn-system.
+
+Before upgrading the longhorn-engine of the volumes, the old replica manager pods are still running and serving the backend data path of the longhorn volumes. After upgrading, the replicas will be launched in new replica manager pods, and the old ones will be cleaned up.
+
+To keep away from the interruption of data IO, the services in engine instance manager pods cannot be switched to the new ones when the volumes are still attached. Therefore, the user will see the old pods are still running after the upgrade process. After detaching the volumes, the old engine manager pods will be removed automatically, if there is no volume engine pointing to the old engine managers.
+
+The user can check which volumes are still pointing to the old engine manager pods from `Volume > Name > Volume Details > Instance Manager` on Longhorn UI.
+
+
```

Comment:
It's nice (but not required) to add a section `Tip` after explaination, which tells users how to let Longhorn clean up all old instance managers. (By detaching all old volumes & the volume performed live upgrade.)